src/marqo/tensor_search/web/api_utils.py

```diff
@@ -161,19 +161,36 @@ def add_docs_params_orchestrator(index_name: str, body: Union[AddDocsBodyParams,
 
         mappings = body.mappings
         non_tensor_fields = body.nonTensorFields
+        tensor_fields = body.tensorFields
         use_existing_tensors = body.useExistingTensors
         model_auth = body.modelAuth
         image_download_headers = body.imageDownloadHeaders
 
+        if tensor_fields and non_tensor_fields:
+            raise BadRequestError('Cannot provide nonTensorFields when tensorFields is defined. '
+                                  'nonTensorFields has been deprecated and will be removed in Marqo 2.0.0. '
+                                  'Its use is discouraged.')
+
+        if not tensor_fields and not non_tensor_fields:
+            raise BadRequestError('You must include the tensorFields parameter. '
+                                  'Use tensorFields=[] to index for lexical-only search.')
```

Comment:
The pattern we are moving to now is to relegate validation to move validation to AddDocsParams validation methods

Reply:
Good error messages! 

Nit on the wording: 
```
You must include the tensorFields parameter.
````
 -> 
```
Required parameter `tensorFields` is missing from the request body.
```

Reply:
Changed